feat(sunburst): Added some options to adjust Sunburst chart view: innerRadius, outerRadius, donut, showNulls

bwa84 opened the pull request:
### SUMMARY
Added some options to adjust Sunburst chart:

- showNulls: allow to show / hide slices with null value
- donut: allow to switch between donut / circle view (donut was fixed before)
- innerRadius: allow to adjust inner radius when donut is checked
- outerRadius: allow to adjust outer radius

### AFTER SCREENSHOT / VIDEO
![obraz](https://github.com/user-attachments/assets/fdfe8483-f036-4ce1-b298-b47c922172a9)

https://github.com/user-attachments/assets/959b022e-2f0b-4537-abc6-c66c435b6ba7



### TESTING INSTRUCTIONS

- create or edit sunburst chart
- switch to Customize control page
- play with "Donut", "Inner Radius", "Outer Radius" & "Show <NULL> values" options

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

Commit:
Added some options to set sunburst view

Commit:
adhusted defaults & typo fix

rusackas commented:
After screenshots (or video) of the controls in effect might help. I've started CI, and will hopefully remember to come back and spin up a test deployment if it passes.

bwa84 commented:
> After screenshots (or video) of the controls in effect might help. I've started CI, and will hopefully remember to come back and spin up a test deployment if it passes.

After screenshot was added previously, now I added video also.

bwa84 commented:
Found bug with cross-filtering. Converting to draft for now.

bwa84 commented:
Little hint for beginner needed.
Bug that I found is not connected with this PR (checked in master & v4.0.1 release).
Can I fix it in this PR or should I fix it in a separate one?

villebro commented:
> Little hint for beginner needed. Bug that I found is not connected with this PR (checked in master & v4.0.1 release). Can I fix it in this PR or should I fix it in a separate one?

@bwa84 please feel free to address the bug in this PR if it seems doable as a bycatch fix. But if it's a substantial change, having a dedicated PR for it could be advisable.

Commit:
Fix bug with cross-filter with >2 sunburst rings.

Fixed bug when only 1st and top level of sunburst can emit cross-filter (when >2 levels). Selecting other ended with error.

villebro commented on superset-frontend/plugins/plugin-chart-echarts/src/Sunburst/types.ts:
```diff
@@ -52,6 +52,10 @@ export const DEFAULT_FORM_DATA: Partial<EchartsSunburstFormData> = {
   labelType: EchartsSunburstLabelType.Key,
   showLabels: false,
   dateFormat: 'smart_date',
+  innerRadius: 30,
+  outerRadius: 70,
+  donut: true,
+  showNulls: true,
```
This will change how all current sunburst charts are rendered. While I'm not strongly against/for this, I'm not sure we want to forcefully change all current charts. For this reason I would recommend changing the defaults to align with the current behavior, and defer changing the defaults to a later date when there's been a decision to to change the defaults. Maybe we could discuss changing the defaults for the pie chart and sunburst chart in a forthcoming town hall? CC @rusackas @michael-s-molina

bwa84 replied:
Default values were set with the same values that were hardcoded (innerRadius = 30, outerRadius = 70) or to keep current default behavior (donut = true, showNulls = true).

rusackas commented:
Needs conflict resolution, so I'll just convert this to draft. If it can be brought up to a mergeable state, just mark it as ready for review!
